types/conf.go

```diff
@@ -122,23 +126,30 @@ func LoadDelegateNetConf(bytes []byte, net *NetworkSelectionElement, deviceID st
 		if net.InfinibandGUIDRequest != "" {
 			delegateConf.InfinibandGUIDRequest = net.InfinibandGUIDRequest
 		}
+		if net.DeviceID != "" {
+			if deviceID != "" {
+				logging.Debugf("Warning: Both RuntimeConfig and ResourceMap provide deviceID. Ignoring RuntimeConfig")
+			} else {
+				delegateConf.DeviceID = net.DeviceID
+			}
+		}
 	}
 
 	delegateConf.Bytes = bytes
 
 	return delegateConf, nil
 }
 
-// MergeCNIRuntimeConfig creates CNI runtimeconfig from delegate
-func MergeCNIRuntimeConfig(runtimeConfig *RuntimeConfig, delegate *DelegateNetConf) *RuntimeConfig {
-	logging.Debugf("MergeCNIRuntimeConfig: %v %v", runtimeConfig, delegate)
+// mergeCNIRuntimeConfig creates CNI runtimeconfig from delegate
+func mergeCNIRuntimeConfig(runtimeConfig *RuntimeConfig, delegate *DelegateNetConf) *RuntimeConfig {
```

Comment:
Just making sure I understand this, but, you're making this a private function in favor of the `types.CreateCNIRuntimeConf` where it's called from multus.go?

Reply:
Yes making private. All cases in multus.go that `mergeCNIRuntimeConfig` was called, it was followed by a call to `CreateCNIRuntimeConf`. So `mergeCNIRuntimeConfig` is now called from `CreateCNIRuntimeConf` and no longer called from multus.go.